src/platform/redox/mod.rs

```diff
@@ -0,0 +1,554 @@
+                                window_id: ::WindowId(WindowId),
+                                event: ::WindowEvent::MouseInput {
+                                    device_id: ::DeviceId(DeviceId),
+                                    state: btn_state,
+                                    button: btn,
+                                    modifiers: state.modifiers(),
+                                }
+                            });
+                        }
+                    },
+                    EventOption::Scroll(event) => callback(::Event::WindowEvent {
+                        window_id: ::WindowId(WindowId),
+                        event: ::WindowEvent::MouseWheel {
+                            device_id: ::DeviceId(DeviceId),
+                            delta: ::MouseScrollDelta::PixelDelta((event.x, event.y).into()),
+                            phase: ::TouchPhase::Moved,
+                            modifiers: state.modifiers(),
+                        }
+                    }),
+                    EventOption::Quit(_event) => callback(::Event::WindowEvent {
+                        window_id: ::WindowId(WindowId),
+                        event: ::WindowEvent::CloseRequested,
+                    }),
+                    EventOption::Focus(event) => callback(::Event::WindowEvent {
+                        window_id: ::WindowId(WindowId),
+                        event: ::WindowEvent::Focused(event.focused),
+                    }),
+                    EventOption::Move(event) => callback(::Event::WindowEvent {
+                        window_id: ::WindowId(WindowId),
+                        event: ::WindowEvent::Moved((event.x, event.y).into()),
+                    }),
+                    EventOption::Resize(event) => callback(::Event::WindowEvent {
+                        window_id: ::WindowId(WindowId),
+                        event: ::WindowEvent::Resized((event.width, event.height).into()),
+                    }),
+                    _ => ()
+                }
+            }
+        }
+    }
+
+    pub fn run_forever<F>(&mut self, mut callback: F)
+        where F: FnMut(::Event) -> ::ControlFlow,
+    {
+        // Yeah that's a very bad implementation.
+        loop {
+            let mut control_flow = ::ControlFlow::Continue;
+            self.poll_events(|e| {
+                if let ::ControlFlow::Break = callback(e) {
+                    control_flow = ::ControlFlow::Break;
+                }
+            });
+            if let ::ControlFlow::Break = control_flow {
+                break;
+            }
+            ::std::thread::sleep(::std::time::Duration::from_millis(5));
+        }
+    }
+
+    pub fn create_proxy(&self) -> EventsLoopProxy {
+        EventsLoopProxy
+    }
+}
+
+#[derive(Clone)]
+pub struct EventsLoopProxy;
+
+impl EventsLoopProxy {
+    pub fn wakeup(&self) -> Result<(), ::EventsLoopClosed> {
+        unimplemented!()
```

Comment:
Is it not possible to implement this currently?

Reply:
I was unsure what exactly this is supposed to do. The simpler modules like ios and android do not implement it, and the more complicated modules like windows and x11/wayland are very confusing to read.

With a little more information I am sure I could implement it.

Reply:
Ok. I will work on implementing it